lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -804,165 +723,102 @@ mkTipSyncClient
-                    onInterpreterUpdate interpreter
-
-        W.GenesisParameters
+    let queryParams = do
+            mb <- currentEra >>= \case
+                AnyCardanoEra ShelleyEra ->
+                    LSQry $ QueryAnytimeShelley GetEraStart
+                _ ->
+                    LSQry $ QueryAnytimeMary GetEraStart
+            sp <- byronOrShelleyBased
+                (pure $ W.slottingParameters np)
+                ((slottingParametersFromGenesis . getCompactGenesis)
+                    <$> LSQry Shelley.GetGenesisConfig)
+            pp <- byronOrShelleyBased
+                (protocolParametersFromUpdateState mb
+                    <$> LSQry Byron.GetUpdateInterfaceState)
+                (fromShelleyPParams mb
+                    <$> LSQry Shelley.GetCurrentPParams)
+            return (pp, sp)
+
+    let queryInterpreter = LSQry (QueryHardFork GetInterpreter)
+
+    let W.GenesisParameters
              { getGenesisBlockHash
              } = W.genesisParameters np
-        cfg = codecConfig (W.slottingParameters np)
+    let cfg = codecConfig (W.slottingParameters np)
 
     onTipUpdate' <- debounce @_ @m $ \(era, tip') -> do
         -- FIXME: Store / replace / keep track of the era and make it available
         -- for other functions.
         let tip = castTip tip'
         traceWith tr $ MsgNodeTip $
             fromTip getGenesisBlockHash tip
-        onTipUpdate (era, tip)
-        queryLocalState era (getTipPoint tip)
-        -- NOTE: interpeter is updated every block. This is far more often than
-        -- necessary.
-        queryInterpreter (getTipPoint tip)
 
-    pure $ nodeToClientProtocols (const $ return $ NodeToClientProtocols
-        { localChainSyncProtocol =
-            let
-                codec = cChainSyncCodec $ codecs cfg
-            in
-            InitiatorProtocolOnly $ MuxPeerRaw
-                $ \channel -> runPeer nullTracer codec channel
-                $ chainSyncClientPeer
-                $ chainSyncFollowTip toCardanoEra (curry onTipUpdate')
+        atomically $ writeTVar tipVar (era, tip)
 
-        , localTxSubmissionProtocol =
-            let
-                tr' = contramap MsgTxSubmission tr
-                codec = cTxSubmissionCodec $ serialisedCodecs cfg
-            in
-            InitiatorProtocolOnly $ MuxPeerRaw
-                $ \channel -> runPeer tr' codec channel
-                $ localTxSubmissionClientPeer
-                $ localTxSubmission localTxSubmissionQ
+        -- NOTE: If these queries are not sent with sendAsync, we'd cause
+        -- deadlock, as the local state query client might need to waiy for the
+        -- tip client to find a new tip.
+        --
+        -- NOTE 2: interpeter is updated every block. This is far more often than
+        -- necessary.
+        sendAsync localStateQueryQ $ SomeLSQ queryParams onPParamsUpdate'
+        sendAsync localStateQueryQ $ SomeLSQ queryInterpreter onInterpreterUpdate
```

Comment:
Arguably, we only need those once per epoch.

Reply:
It would be cool to add some logic for that, but it would probably be non-trivial.

Reply:
For instance, we can't observe the node tip `STM` and filter out the epoch-changes, since we're not guaranteed to see all intermediate states with `STM`.